test/parallel/test-async-local-storage-http-multiclients.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+const common = require('../common');
+
+const assert = require('assert');
+const { AsyncLocalStorage } = require('async_hooks');
+const http = require('http');
+const cls = new AsyncLocalStorage();
+const NUM_CLIENTS = 10;
+
+// Run multiple clients that receive data from a server
+// in multiple chunks, in a single non-closure function.
+// Use the AsyncLocalStorage (ALS) APIs to maintain the context
+// and data download. Make sure that individual clients
+// receive their respective data, with no conflicts.
+
+// Set up a server, that sends large buffers of data, filled
+// with cardinal numbers, increasing per request
+let index = 0;
+const server = http.createServer((q, r) => {
+  // Send a large chunk as response, otherwise the data
+  // may be coalesced, and the callback in the client
+  // may be called only once, defeating the purpose of test
+  r.end((index++).toString().repeat(1024 * 1024));
```

Comment:
This is going to break if `NUM_CLIENTS` is set to a value > 10. Can you either add a comment that warns about that, or (preferably) make the test more robust by not relying on the fact that `index < 10` here?

Reply:
Added `%10` to the current index, this solves the problem if `NUM_CLIENTS` value is >10.